fluent-bit-init-process.go

```diff
@@ -0,0 +1,402 @@
+		// if not specify region here, missingregion error will raise when get bucket location
+		Region: aws.String(region),
+	})))
+
+	exists3Client = true
+}
+
+// create a S3 Downloader
+func createS3Downloader(bucketRegion string) S3Downloader {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(bucketRegion)},
+	)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalln("[FluentBit Init Process] Cannot creat a new session")
+	}
+
+	// need to specify session region!
+	s3Downloader := s3manager.NewDownloader(sess)
+	return s3Downloader
+}
+
+// download file from S3 and store
+func downloadS3ConfigFile(s3Downloader S3Downloader, s3FilePath, bucketName string) {
+
+	s3FileName := strings.SplitN(s3FilePath, "/", -1)
+	fileFromS3 := createFile(s3FileFolderPath+"/"+s3FileName[len(s3FileName)-1], false)
+	defer fileFromS3.Close()
+
+	_, err := s3Downloader.Download(fileFromS3,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String(s3FilePath),
+		})
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot download %s from s3\n", s3FileName)
+	}
+}
+
+// process S3 config files user specified
+func processS3ConfigFiles(folderPath string) {
+
+	fileInfos, err := ioutil.ReadDir(folderPath)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Unable to read config files in %s folder\n", folderPath)
+	}
+
+	for _, file := range fileInfos {
+		filePath := "/" + folderPath + "/" + file.Name()
+		processConfigFile(filePath)
+	}
+}
+
+// use @INCLUDE to add config files to the main config file
+func writeInclude(configFilePath, mainConfigFilePath string) {
+
+	mainConfigFile := openFile(mainConfigFilePath)
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + configFilePath + "\n"
+	_, err := mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+// change the cammand if needed.
+// add modified command to the init_process_entrypoint.sh
+func updateCommand(parserFilePath string) {
+	fluentBitCommand = fluentBitCommand + " -R " + parserFilePath
+	logrus.Infoln("[FluentBit Init Process] Command is change to -> " + fluentBitCommand)
+}
+
+// change the fluent-bit-invoker.sh
+// which will declare ECS Task Metadata as environment variables
+// and finally invoke Fluent Bit
+func modifyInvokerFile(filePath string) {
+
+	invokerFile := openFile(filePath)
+	defer invokerFile.Close()
+
+	_, err := invokerFile.WriteString(fluentBitCommand)
+	if err != nil {
+		logrus.Errorln(err)
+		logrus.Fatalf("[FluentBit Init Process] Cannot write %s in fluent-bit-invoker.sh\n", fluentBitCommand)
+	}
+}
+
+// create a file, when flag is true, the file will be closed automatically after creation
+func createFile(filePath string, AutoClose bool) *os.File {
```

Comment:
when autoclose will be set to false?

Reply:
https://github.com/aws/aws-for-fluent-bit/pull/379/files#diff-a24b896daae00708d98524f520e1d75af05aea6586bd9e70568e44197652681eR273

Reply:
then we will not close it after downloading content?